python/ray/util/sgd/torch/training_operator.py

```diff
@@ -109,13 +146,197 @@ def _set_timers(self, timers):
         self.timers = timers
 
     def setup(self, config):
-        """Override this method to implement custom operator setup.
+        """Override this method to implement custom operator setup. You must
+        call self.register here to register training components with Ray SGD.
 
         Args:
             config (dict): Custom configuration value to be passed to
                 all creator and operator constructors. Same as ``self.config``.
         """
-        pass
+        raise NotImplementedError
+
+    def register(self,
+                 *,
+                 models,
+                 optimizers,
+                 train_loader,
+                 validation_loader,
+                 criterion=None,
+                 schedulers=None):
+        """Registers parameters with Ray SGD. Also sets up necessary
+        training components (Cuda, DDP, Distributed Sampler, Fp16).
+        You do not need to handle GPU/devices in this function; ``Ray SGD``
+        will do that for you.
+
+        If more than one model, optimizer, or scheduler is passed in,
+        you should implement your own custom training loop.
+
+        .. code-block:: python
+
+            @override(TrainingOperator)
```

Comment:
Maybe leave this out or use a comment instead?

Reply:
Removed.